### ISSUE TITLE
Add pings for VS Code extension

### ISSUE THREAD

**Issue by tjkandala** [sourcegraph/sourcegraph#29124]
Add pings specified in [RFC 547](https://docs.google.com/document/d/1YzDrXEhpa0dvE4On1S672w8l01EsLcXhLsJrLOgcQKQ/edit#).

**Comment by muratsu** [comment]
Fyi: https://github.com/sourcegraph/sourcegraph/issues/27861

**Comment by jjinnii** [comment]
In addition to pings outlined in [RFC 547](https://docs.google.com/document/d/1YzDrXEhpa0dvE4On1S672w8l01EsLcXhLsJrLOgcQKQ/edit#), the below metrics are outlined in Figma: 

**Pings requiring UTM parameters:** 
* ["Create an account"](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3639) in sidebar: Count of # of times this button was shown to users. Count of # of times this button was clicked by users. 
* ["Create a free account"](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3487) in banner: Count of # of times this banner was shown to users. Count of # of times this button was clicked by users. 
* ["Sign up for Sourcegraph" in "Monitor" popover](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3585): Count of # of times "Monitor" was clicked by unauthenticated users, # of times the popover was seen by users. Count of # of times this button was clicked by users. 
* ["Sign up for Sourcegraph"](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3585) in "Save Search" popover: Count of # of times "Save Search" was clicked by unauthenticated users, # of times the popover was seen by users. Count of # of times this button was clicked by users. 

For these pings, UTM parameter are required to track for users who clicked on above buttons, how many users initiated signup, successfully signed up, then saved repos post sign-up. For reference, the goal is to create a [funnel conversion view ](https://analytics.amplitude.com/sourcegraph/notebook/yewij8w/breadcrumbs/chart/09hgzfn?source=sidebar).



**Usage oriented pings:** 
 * [Links under Learn more in sidebar clicked](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1396%3A3362)
 * [Search examples clicked](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3661)
 * [Watch & Learn Video clicked](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3617)
 * Filter features in sidebar used: [Search types](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1390%3A3543), [Dynamic Filters](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1395%3A3304),[ Repositories](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1396%3A3338), Search Reference

**Comment by tjkandala** [comment]
Note from review meeting: let's track existing extension feature usage too (e.g. open in Sourcegraph)

**Comment by jjinnii** [comment]
FYI added a new metric request for the "Share" link button, documented[ here ](https://www.figma.com/file/EXXvfxVwtCIsrbQez4rPlF/VS-Code-Extension-MLP?node-id=1384%3A3428)in Figma.

**Comment by jjinnii** [comment]
I can't believe I missed this during planning... Did we get this done in time for the 3.37 cut? e.g. will we be able to access Feb data for customers who upgraded? @tjkandala @abeatrix

**Comment by abeatrix** [comment]
[RFC 577](https://docs.google.com/document/d/1fdathYEEOUgyLbnpEKV7g_7qqHfbvftCGVQt_mXyqmI/edit#)

**Comment by abeatrix** [comment]
I have updated the payload example for the Ide extensions pings to: 
```
{
    "ide_extensions_usage": {
        "IDEs": [
            {
                "IdeKind": "vscode",
                "Day": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    }
                },
                "Week": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    }
                },
                "Month": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    },
                    "UserStateCount": {
                        "Installed": "NUMBER",
                        "Uninstalled": "NUMBER"
                    },
                    "RedirectCount": "NUMBER"
                }
            },
            {
                "IdeKind": "jetbrains",
                "Day": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    }
                },
                "Week": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    }
                },
                "Month": {
                    "StartTime": "DATE_TIME",
                    "UserCount": "NUMBER",
                    "SearchPerformed": {
                        "TotalCount": "NUMBER",
                        "UniqueCount": "NUMBER"
                    },
                    "UserStateCount": {
                        "Installed": "NUMBER",
                        "Uninstalled": "NUMBER"
                    },
                    "RedirectCount": "NUMBER"
                }
            },
        ]
    }
}
```

Currently, we are getting the following:  
- `Monthly Installs` numbers from users whose oldest ide-related event is within the current month.
- `Monthly uninstalls` numbers from users who have the `IDEUninstalled` event (To be implemented in VSCE v2.2.0)

IDE extensions should log events under `source = IDEEXTENSION` for instances above v3.37.0.

**Comment by abeatrix** [comment]
Update:
- PR to add pings for IDE extensions has been completed: https://github.com/sourcegraph/sourcegraph/pull/32000
  - Currently waiting for Business Ops to approve [the schema PR](https://github.com/sourcegraph/analytics/pull/368) before it can be merged
- PR to add IDE extensions usage pings to big query schema has been completed: https://github.com/sourcegraph/analytics/pull/368
  - Currently waiting for Business Ops to approve the PR before it can be merged
- PR to log the correct event names in VSCE has been completed: https://github.com/sourcegraph/sourcegraph/pull/32087
  - need to release this build once it is merged
